hooks/useResearchOutputTable.ts

```diff
@@ -0,0 +1,724 @@
+  'Metadata Standards',
+  'Licenses',
+  'Initial Access Levels',
+]);
+
+// Custom hook for research output table
+export const useResearchOutputTable = ({ setHasUnsavedChanges, announce }: { setHasUnsavedChanges: React.Dispatch<React.SetStateAction<boolean>>, announce: (message: string) => void }) => {
+  // States for Research Output table question type
+  // Which fields are expanded for customization
+  const [expandedFields, setExpandedFields] = useState<string[]>(['title', 'outputType']);
+  // Which fields cannot be customized
+  const nonCustomizableFieldIds = ['accessLevels'];
+  // Standard fields for research output questions
+  const [standardFields, setStandardFields] = useState(initialStandardFields);
+  // Additional fields for research output questions
+  const [additionalFields, setAdditionalFields] = useState<AdditionalFieldsType[]>([]);
+  // State for managing custom output types
+  const [newOutputType, setNewOutputType] = useState<OutputTypeInterface>({ type: '', description: '' });
+  // State for managing custom license types
+  const [newLicenseType, setNewLicenseType] = useState<string>('');
+
+
+  // Query request for all licenses
+  const { data: licensesData } = useLicensesQuery();
+
+  // Query request for default research output types
+  const { data: defaultResearchOutputTypesData } = useDefaultResearchOutputTypesQuery();
+
+  // localization keys
+
+  const QuestionAdd = useTranslations('QuestionAdd');
+
+  // Type guard function to check if a field has metaDataConfig
+  const hasMetaDataConfig = (field: StandardField): field is StandardField & { metaDataConfig: MetaDataConfig } => {
+    return field.metaDataConfig !== undefined;
+  };
+
+
+  /**
+   * Build form state for research output table questions.
+   * 
+   * Constructs a table-based question structure conforming to ResearchOutputTableQuestion schema
+   * from @dmptool/types. Each enabled standardField is converted into a table column with
+   * appropriate field types:
+   * 
+   * - title: text field (always required)
+   * - description: textArea with rich text support
+   * - outputType: selectBox with configured options
+   * - dataFlags: boolean fields for sensitive/personal data flags
+   * - repoSelector: repositorySearch with GraphQL query
+   * - metadataStandards: metadataStandardSearch with GraphQL query
+   * - licenses: licenseSearch with GraphQL query (not selectBox)
+   * - accessLevels: selectBox with access level options
+   * - additionalFields: custom text fields
+   * 
+   * This function prepares the user input structure that will be passed to the
+   * researchOutputTable handler in questionTypeHandlers for final validation.
+   */
+  const buildResearchOutputFormState = (parsed: AnyParsedQuestion | null) => {
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+    const columns: any[] = [];
+
+    standardFields.forEach(field => {
+      if (!field.enabled) return;
+
+      switch (field.id) {
+        case 'title':
```

Comment:
It might be better to use the new defaults that we're exporting from the dmptool/types package (e.g.`DefaultTitleQuestion`) it this switch block, or just `DefaultResearchOutputTable` which should return everything you need I think

I think that could be done in a seperate follow up ticket though